pkg/splunk/enterprise/configuration_test.go

```diff
@@ -845,3 +846,71 @@ func TestAreRemoteVolumeKeysChanged(t *testing.T) {
 		t.Errorf("Empty volume should not report a key change")
 	}
 }
+
+func TestAddStorageVolumes(t *testing.T) {
+	labels := make(map[string]string)
+	var replicas int32 = 1
+
+	// Create CR
+	cr := enterprisev1.ClusterMaster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "CM",
+			Namespace: "test",
+		},
+	}
+
+	// create statefulset configuration
+	statefulSet := &appsv1.StatefulSet{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "StatefulSet",
+			APIVersion: "apps/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-statefulset",
+			Namespace: cr.GetNamespace(),
+		},
+		Spec: appsv1.StatefulSetSpec{
+			Replicas: &replicas,
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Image: "test",
+							Name:  "splunk",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Define valid common spec
+	spec := &enterprisev1.CommonSplunkSpec{}
+	err := addStorageVolumes(&cr, spec, statefulSet, labels)
+	if err != nil {
+		t.Errorf("Unable to add storage volumes, error: %s", err.Error())
+	}
```

Comment:
Any tests where we confirm if the (default?) Volumes are indeed mounted? Or can we peek into statefulset spec template to confirm this? Looks like at this point we just go by whether the API is returning success or error, without checking if the intended effect has been taken